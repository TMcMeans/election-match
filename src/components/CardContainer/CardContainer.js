import React from 'react';

export const CardContainer = ({ electionData, error }) => {
  return (
    <div className="card-container">
      <h1>Election Data</h1>
    </div>
  );
};
