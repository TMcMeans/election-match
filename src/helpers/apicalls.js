export const fetchElections = async url => {
  let response = await fetch(url, {
    method: 'GET',
    headers: {
      Accept: 'application/json'
    }
  });

  if (response.ok) {
    let electionData = await response.json();
    return electionData;
  } else {
    console.log(`${response.status}: ${response.statusText}`);
  }
};

export const postAddress = async ocd_id => {
  let response = await fetch('http://localhost:3000/api/search', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      ocd_id
    })
  });

  if (response.ok) {
    console.log('sucessfully sent user address to server');
  } else {
    console.log(`${response.status}: ${response.statusText}`);
  }
};
