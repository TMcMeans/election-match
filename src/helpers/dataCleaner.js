export const cleanAddress = uncleanAddress => {
  let place = uncleanAddress.city.toLowerCase();
  place.replace(' ', '_');
  let state = uncleanAddress.state.toLowerCase();

  return { place, state };
};
