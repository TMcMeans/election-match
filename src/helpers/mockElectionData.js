export const mockElectionData = [
  {
    'district-divisions': [
      {
        'ocd-id': 'ocd-division/country:us/state:ma/place:rochester',
        'voter-registration-authority-level': 'municipal',
        'election-authority-level': 'municipal',
        'voting-methods': [
          {
            primary: false,
            type: 'early-voting',
            'excuse-required': false
          },
          {
            primary: true,
            instructions: {
              'voting-id':
                "You may be asked to show identification if: you are voting for the first-time in Massachusetts, you are an inactive voter, you are casting a provisional or challenged ballot, or if the poll worker has a reasonable suspicion that leads them to request identification. Acceptable forms include (must show your name and address): a MA driver's license or MA-issued ID card; recent utility bill; rent receipt; signed lease; a copy of a voter registration affidavit; or any other printed identification which contains the voter's name and address.\n\nVoters without ID: If you're a first-time voter who is unable to present ID when you check in, you may vote a provisional ballot and return with acceptable ID by close of polls. If you're asked for ID for any other reason, and are not able to present ID in such a situation, you must still be permitted to vote; however, your ballot must be challenged. Your ballot will be cast normally, and will only be re-examined in the case of a recount, court order, or audit."
            },
            type: 'in-person',
            'excuse-required': false
          },
          {
            primary: false,
            type: 'by-mail',
            'excuse-required': true,
            'ballot-request-deadline-received': '2019-04-09T00:00:00Z',
            'acceptable-forms': [
              {
                name: 'ma_absentee'
              }
            ]
          }
        ],
        'voter-registration-methods': [
          {
            instructions: {
              registration:
                "You should know: you need a Massachusetts ID to use Massachusetts's online voter registration system. If you don't have a Massachusetts-issued ID, you can still register to vote by mail."
            },
            type: 'online',
            'supports-iframe': true,
            'deadline-online': '2019-03-21T00:00:00Z',
            url:
              'https://www.sec.state.ma.us/OVR/Pages/MinRequirements.aspx?RMVId=True'
          },
          {
            'deadline-postmarked': '2019-03-21T00:00:00Z',
            instructions: {
              signature:
                'To register in Massachusetts you must: \nbe a citizen of the United States \nbe a resident of Massachusetts \nbe 18 years old on or before the next election \nnot have been convicted of corrupt practices in respect to elections \nnot be under guardianship with respect to voting \nnot be currently incarcerated for a felony conviction',
              idnumber:
                'Federal law requires that you provide your driver\'s license number to register to vote. If you do not have a current and valid Massachusetts driver\'s license, you must provide the last four digits of your Social Security number. If you have neither, you must write "NONE" in the box.'
            },
            type: 'by-mail',
            'acceptable-forms': [
              {
                name: 'nvrf'
              }
            ]
          }
        ]
      }
    ],
    website: 'https://www.townofrochestermass.com/townclerk.html',
    'polling-place-url':
      'https://www.sec.state.ma.us/wheredoivotema//bal/myelectioninfo.aspx',
    date: '2019-04-10T00:00:00Z',
    population: 5575,
    'polling-place-url-shortened': 'https://tvote.org/2v2xAee',
    description: 'Rochester Town Election',
    id: '5c5c6acf-4a72-49be-a122-ec9cb673a4ab'
  }
];
