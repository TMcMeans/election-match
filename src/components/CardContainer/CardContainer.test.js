import React from 'react';
import { shallow } from 'enzyme';
import { mockElectionData } from '../../helpers/mockElectionData';

import { CardContainer } from './CardContainer';

describe('CardContainer', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<CardContainer electionData={mockElectionData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
