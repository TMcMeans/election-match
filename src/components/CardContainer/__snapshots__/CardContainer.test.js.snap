// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    electionData={
      Array [
        Object {
          "date": "2019-04-10T00:00:00Z",
          "description": "Rochester Town Election",
          "district-divisions": Array [
            Object {
              "election-authority-level": "municipal",
              "ocd-id": "ocd-division/country:us/state:ma/place:rochester",
              "voter-registration-authority-level": "municipal",
              "voter-registration-methods": Array [
                Object {
                  "deadline-online": "2019-03-21T00:00:00Z",
                  "instructions": Object {
                    "registration": "You should know: you need a Massachusetts ID to use Massachusetts's online voter registration system. If you don't have a Massachusetts-issued ID, you can still register to vote by mail.",
                  },
                  "supports-iframe": true,
                  "type": "online",
                  "url": "https://www.sec.state.ma.us/OVR/Pages/MinRequirements.aspx?RMVId=True",
                },
                Object {
                  "acceptable-forms": Array [
                    Object {
                      "name": "nvrf",
                    },
                  ],
                  "deadline-postmarked": "2019-03-21T00:00:00Z",
                  "instructions": Object {
                    "idnumber": "Federal law requires that you provide your driver's license number to register to vote. If you do not have a current and valid Massachusetts driver's license, you must provide the last four digits of your Social Security number. If you have neither, you must write \\"NONE\\" in the box.",
                    "signature": "To register in Massachusetts you must: 
be a citizen of the United States 
be a resident of Massachusetts 
be 18 years old on or before the next election 
not have been convicted of corrupt practices in respect to elections 
not be under guardianship with respect to voting 
not be currently incarcerated for a felony conviction",
                  },
                  "type": "by-mail",
                },
              ],
              "voting-methods": Array [
                Object {
                  "excuse-required": false,
                  "primary": false,
                  "type": "early-voting",
                },
                Object {
                  "excuse-required": false,
                  "instructions": Object {
                    "voting-id": "You may be asked to show identification if: you are voting for the first-time in Massachusetts, you are an inactive voter, you are casting a provisional or challenged ballot, or if the poll worker has a reasonable suspicion that leads them to request identification. Acceptable forms include (must show your name and address): a MA driver's license or MA-issued ID card; recent utility bill; rent receipt; signed lease; a copy of a voter registration affidavit; or any other printed identification which contains the voter's name and address.

Voters without ID: If you're a first-time voter who is unable to present ID when you check in, you may vote a provisional ballot and return with acceptable ID by close of polls. If you're asked for ID for any other reason, and are not able to present ID in such a situation, you must still be permitted to vote; however, your ballot must be challenged. Your ballot will be cast normally, and will only be re-examined in the case of a recount, court order, or audit.",
                  },
                  "primary": true,
                  "type": "in-person",
                },
                Object {
                  "acceptable-forms": Array [
                    Object {
                      "name": "ma_absentee",
                    },
                  ],
                  "ballot-request-deadline-received": "2019-04-09T00:00:00Z",
                  "excuse-required": true,
                  "primary": false,
                  "type": "by-mail",
                },
              ],
            },
          ],
          "id": "5c5c6acf-4a72-49be-a122-ec9cb673a4ab",
          "polling-place-url": "https://www.sec.state.ma.us/wheredoivotema//bal/myelectioninfo.aspx",
          "polling-place-url-shortened": "https://tvote.org/2v2xAee",
          "population": 5575,
          "website": "https://www.townofrochestermass.com/townclerk.html",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Your results:
        </h2>,
        Array [
          <ElectionCard
            date="2019-04-10T00:00:00Z"
            description="Rochester Town Election"
            district-divisions={
              Array [
                Object {
                  "election-authority-level": "municipal",
                  "ocd-id": "ocd-division/country:us/state:ma/place:rochester",
                  "voter-registration-authority-level": "municipal",
                  "voter-registration-methods": Array [
                    Object {
                      "deadline-online": "2019-03-21T00:00:00Z",
                      "instructions": Object {
                        "registration": "You should know: you need a Massachusetts ID to use Massachusetts's online voter registration system. If you don't have a Massachusetts-issued ID, you can still register to vote by mail.",
                      },
                      "supports-iframe": true,
                      "type": "online",
                      "url": "https://www.sec.state.ma.us/OVR/Pages/MinRequirements.aspx?RMVId=True",
                    },
                    Object {
                      "acceptable-forms": Array [
                        Object {
                          "name": "nvrf",
                        },
                      ],
                      "deadline-postmarked": "2019-03-21T00:00:00Z",
                      "instructions": Object {
                        "idnumber": "Federal law requires that you provide your driver's license number to register to vote. If you do not have a current and valid Massachusetts driver's license, you must provide the last four digits of your Social Security number. If you have neither, you must write \\"NONE\\" in the box.",
                        "signature": "To register in Massachusetts you must: 
be a citizen of the United States 
be a resident of Massachusetts 
be 18 years old on or before the next election 
not have been convicted of corrupt practices in respect to elections 
not be under guardianship with respect to voting 
not be currently incarcerated for a felony conviction",
                      },
                      "type": "by-mail",
                    },
                  ],
                  "voting-methods": Array [
                    Object {
                      "excuse-required": false,
                      "primary": false,
                      "type": "early-voting",
                    },
                    Object {
                      "excuse-required": false,
                      "instructions": Object {
                        "voting-id": "You may be asked to show identification if: you are voting for the first-time in Massachusetts, you are an inactive voter, you are casting a provisional or challenged ballot, or if the poll worker has a reasonable suspicion that leads them to request identification. Acceptable forms include (must show your name and address): a MA driver's license or MA-issued ID card; recent utility bill; rent receipt; signed lease; a copy of a voter registration affidavit; or any other printed identification which contains the voter's name and address.

Voters without ID: If you're a first-time voter who is unable to present ID when you check in, you may vote a provisional ballot and return with acceptable ID by close of polls. If you're asked for ID for any other reason, and are not able to present ID in such a situation, you must still be permitted to vote; however, your ballot must be challenged. Your ballot will be cast normally, and will only be re-examined in the case of a recount, court order, or audit.",
                      },
                      "primary": true,
                      "type": "in-person",
                    },
                    Object {
                      "acceptable-forms": Array [
                        Object {
                          "name": "ma_absentee",
                        },
                      ],
                      "ballot-request-deadline-received": "2019-04-09T00:00:00Z",
                      "excuse-required": true,
                      "primary": false,
                      "type": "by-mail",
                    },
                  ],
                },
              ]
            }
            id="5c5c6acf-4a72-49be-a122-ec9cb673a4ab"
            polling-place-url="https://www.sec.state.ma.us/wheredoivotema//bal/myelectioninfo.aspx"
            polling-place-url-shortened="https://tvote.org/2v2xAee"
            population={5575}
            website="https://www.townofrochestermass.com/townclerk.html"
          />,
        ],
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Your results:",
        },
        "ref": null,
        "rendered": "Your results:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "5c5c6acf-4a72-49be-a122-ec9cb673a4ab",
        "nodeType": "function",
        "props": Object {
          "date": "2019-04-10T00:00:00Z",
          "description": "Rochester Town Election",
          "district-divisions": Array [
            Object {
              "election-authority-level": "municipal",
              "ocd-id": "ocd-division/country:us/state:ma/place:rochester",
              "voter-registration-authority-level": "municipal",
              "voter-registration-methods": Array [
                Object {
                  "deadline-online": "2019-03-21T00:00:00Z",
                  "instructions": Object {
                    "registration": "You should know: you need a Massachusetts ID to use Massachusetts's online voter registration system. If you don't have a Massachusetts-issued ID, you can still register to vote by mail.",
                  },
                  "supports-iframe": true,
                  "type": "online",
                  "url": "https://www.sec.state.ma.us/OVR/Pages/MinRequirements.aspx?RMVId=True",
                },
                Object {
                  "acceptable-forms": Array [
                    Object {
                      "name": "nvrf",
                    },
                  ],
                  "deadline-postmarked": "2019-03-21T00:00:00Z",
                  "instructions": Object {
                    "idnumber": "Federal law requires that you provide your driver's license number to register to vote. If you do not have a current and valid Massachusetts driver's license, you must provide the last four digits of your Social Security number. If you have neither, you must write \\"NONE\\" in the box.",
                    "signature": "To register in Massachusetts you must: 
be a citizen of the United States 
be a resident of Massachusetts 
be 18 years old on or before the next election 
not have been convicted of corrupt practices in respect to elections 
not be under guardianship with respect to voting 
not be currently incarcerated for a felony conviction",
                  },
                  "type": "by-mail",
                },
              ],
              "voting-methods": Array [
                Object {
                  "excuse-required": false,
                  "primary": false,
                  "type": "early-voting",
                },
                Object {
                  "excuse-required": false,
                  "instructions": Object {
                    "voting-id": "You may be asked to show identification if: you are voting for the first-time in Massachusetts, you are an inactive voter, you are casting a provisional or challenged ballot, or if the poll worker has a reasonable suspicion that leads them to request identification. Acceptable forms include (must show your name and address): a MA driver's license or MA-issued ID card; recent utility bill; rent receipt; signed lease; a copy of a voter registration affidavit; or any other printed identification which contains the voter's name and address.

Voters without ID: If you're a first-time voter who is unable to present ID when you check in, you may vote a provisional ballot and return with acceptable ID by close of polls. If you're asked for ID for any other reason, and are not able to present ID in such a situation, you must still be permitted to vote; however, your ballot must be challenged. Your ballot will be cast normally, and will only be re-examined in the case of a recount, court order, or audit.",
                  },
                  "primary": true,
                  "type": "in-person",
                },
                Object {
                  "acceptable-forms": Array [
                    Object {
                      "name": "ma_absentee",
                    },
                  ],
                  "ballot-request-deadline-received": "2019-04-09T00:00:00Z",
                  "excuse-required": true,
                  "primary": false,
                  "type": "by-mail",
                },
              ],
            },
          ],
          "id": "5c5c6acf-4a72-49be-a122-ec9cb673a4ab",
          "polling-place-url": "https://www.sec.state.ma.us/wheredoivotema//bal/myelectioninfo.aspx",
          "polling-place-url-shortened": "https://tvote.org/2v2xAee",
          "population": 5575,
          "website": "https://www.townofrochestermass.com/townclerk.html",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Your results:
          </h2>,
          Array [
            <ElectionCard
              date="2019-04-10T00:00:00Z"
              description="Rochester Town Election"
              district-divisions={
                Array [
                  Object {
                    "election-authority-level": "municipal",
                    "ocd-id": "ocd-division/country:us/state:ma/place:rochester",
                    "voter-registration-authority-level": "municipal",
                    "voter-registration-methods": Array [
                      Object {
                        "deadline-online": "2019-03-21T00:00:00Z",
                        "instructions": Object {
                          "registration": "You should know: you need a Massachusetts ID to use Massachusetts's online voter registration system. If you don't have a Massachusetts-issued ID, you can still register to vote by mail.",
                        },
                        "supports-iframe": true,
                        "type": "online",
                        "url": "https://www.sec.state.ma.us/OVR/Pages/MinRequirements.aspx?RMVId=True",
                      },
                      Object {
                        "acceptable-forms": Array [
                          Object {
                            "name": "nvrf",
                          },
                        ],
                        "deadline-postmarked": "2019-03-21T00:00:00Z",
                        "instructions": Object {
                          "idnumber": "Federal law requires that you provide your driver's license number to register to vote. If you do not have a current and valid Massachusetts driver's license, you must provide the last four digits of your Social Security number. If you have neither, you must write \\"NONE\\" in the box.",
                          "signature": "To register in Massachusetts you must: 
be a citizen of the United States 
be a resident of Massachusetts 
be 18 years old on or before the next election 
not have been convicted of corrupt practices in respect to elections 
not be under guardianship with respect to voting 
not be currently incarcerated for a felony conviction",
                        },
                        "type": "by-mail",
                      },
                    ],
                    "voting-methods": Array [
                      Object {
                        "excuse-required": false,
                        "primary": false,
                        "type": "early-voting",
                      },
                      Object {
                        "excuse-required": false,
                        "instructions": Object {
                          "voting-id": "You may be asked to show identification if: you are voting for the first-time in Massachusetts, you are an inactive voter, you are casting a provisional or challenged ballot, or if the poll worker has a reasonable suspicion that leads them to request identification. Acceptable forms include (must show your name and address): a MA driver's license or MA-issued ID card; recent utility bill; rent receipt; signed lease; a copy of a voter registration affidavit; or any other printed identification which contains the voter's name and address.

Voters without ID: If you're a first-time voter who is unable to present ID when you check in, you may vote a provisional ballot and return with acceptable ID by close of polls. If you're asked for ID for any other reason, and are not able to present ID in such a situation, you must still be permitted to vote; however, your ballot must be challenged. Your ballot will be cast normally, and will only be re-examined in the case of a recount, court order, or audit.",
                        },
                        "primary": true,
                        "type": "in-person",
                      },
                      Object {
                        "acceptable-forms": Array [
                          Object {
                            "name": "ma_absentee",
                          },
                        ],
                        "ballot-request-deadline-received": "2019-04-09T00:00:00Z",
                        "excuse-required": true,
                        "primary": false,
                        "type": "by-mail",
                      },
                    ],
                  },
                ]
              }
              id="5c5c6acf-4a72-49be-a122-ec9cb673a4ab"
              polling-place-url="https://www.sec.state.ma.us/wheredoivotema//bal/myelectioninfo.aspx"
              polling-place-url-shortened="https://tvote.org/2v2xAee"
              population={5575}
              website="https://www.townofrochestermass.com/townclerk.html"
            />,
          ],
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your results:",
          },
          "ref": null,
          "rendered": "Your results:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "5c5c6acf-4a72-49be-a122-ec9cb673a4ab",
          "nodeType": "function",
          "props": Object {
            "date": "2019-04-10T00:00:00Z",
            "description": "Rochester Town Election",
            "district-divisions": Array [
              Object {
                "election-authority-level": "municipal",
                "ocd-id": "ocd-division/country:us/state:ma/place:rochester",
                "voter-registration-authority-level": "municipal",
                "voter-registration-methods": Array [
                  Object {
                    "deadline-online": "2019-03-21T00:00:00Z",
                    "instructions": Object {
                      "registration": "You should know: you need a Massachusetts ID to use Massachusetts's online voter registration system. If you don't have a Massachusetts-issued ID, you can still register to vote by mail.",
                    },
                    "supports-iframe": true,
                    "type": "online",
                    "url": "https://www.sec.state.ma.us/OVR/Pages/MinRequirements.aspx?RMVId=True",
                  },
                  Object {
                    "acceptable-forms": Array [
                      Object {
                        "name": "nvrf",
                      },
                    ],
                    "deadline-postmarked": "2019-03-21T00:00:00Z",
                    "instructions": Object {
                      "idnumber": "Federal law requires that you provide your driver's license number to register to vote. If you do not have a current and valid Massachusetts driver's license, you must provide the last four digits of your Social Security number. If you have neither, you must write \\"NONE\\" in the box.",
                      "signature": "To register in Massachusetts you must: 
be a citizen of the United States 
be a resident of Massachusetts 
be 18 years old on or before the next election 
not have been convicted of corrupt practices in respect to elections 
not be under guardianship with respect to voting 
not be currently incarcerated for a felony conviction",
                    },
                    "type": "by-mail",
                  },
                ],
                "voting-methods": Array [
                  Object {
                    "excuse-required": false,
                    "primary": false,
                    "type": "early-voting",
                  },
                  Object {
                    "excuse-required": false,
                    "instructions": Object {
                      "voting-id": "You may be asked to show identification if: you are voting for the first-time in Massachusetts, you are an inactive voter, you are casting a provisional or challenged ballot, or if the poll worker has a reasonable suspicion that leads them to request identification. Acceptable forms include (must show your name and address): a MA driver's license or MA-issued ID card; recent utility bill; rent receipt; signed lease; a copy of a voter registration affidavit; or any other printed identification which contains the voter's name and address.

Voters without ID: If you're a first-time voter who is unable to present ID when you check in, you may vote a provisional ballot and return with acceptable ID by close of polls. If you're asked for ID for any other reason, and are not able to present ID in such a situation, you must still be permitted to vote; however, your ballot must be challenged. Your ballot will be cast normally, and will only be re-examined in the case of a recount, court order, or audit.",
                    },
                    "primary": true,
                    "type": "in-person",
                  },
                  Object {
                    "acceptable-forms": Array [
                      Object {
                        "name": "ma_absentee",
                      },
                    ],
                    "ballot-request-deadline-received": "2019-04-09T00:00:00Z",
                    "excuse-required": true,
                    "primary": false,
                    "type": "by-mail",
                  },
                ],
              },
            ],
            "id": "5c5c6acf-4a72-49be-a122-ec9cb673a4ab",
            "polling-place-url": "https://www.sec.state.ma.us/wheredoivotema//bal/myelectioninfo.aspx",
            "polling-place-url-shortened": "https://tvote.org/2v2xAee",
            "population": 5575,
            "website": "https://www.townofrochestermass.com/townclerk.html",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
